### PR TITLE
feature: pipeline support buildkit

### PR DESCRIPTION
**Pull request by iutx**
#### What type of this PR

Add one of the following kinds:
/kind feature

#### What this PR does / why we need it:
pipeline support buildkit

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @luobily @sixther-dc 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |  pipeline support buildkit            |
| 🇨🇳 中文    |   pipeline 支持 buildkit          |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/plugins/k8sjob/k8sjob.go [hunk 1]
```diff
@@ -499,6 +501,47 @@ func (k *K8sJob) generateKubeJob(specObj interface{}) (*batchv1.Job, error) {
 		return nil, errors.Errorf("failed to get cluster info, clusterName: %s, (%v)", k.clusterName, err)
 	}
 
+	if clusterInfo[apistructs.BuildkitEnable] == "true" {
```
Comment (sfwn) [inline modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/plugins/k8sjob/k8sjob.go:504]: bad string comparation. use strconv.parsebool instead.

File: modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/plugins/k8sjob/k8sjob.go [hunk 2]
```diff
@@ -499,6 +501,47 @@ func (k *K8sJob) generateKubeJob(specObj interface{}) (*batchv1.Job, error) {
...
+		delete(clusterInfo, apistructs.BuildkitEnable)
```
Comment (sfwn) [inline modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/plugins/k8sjob/k8sjob.go:505]: any concurrent map rw issue?

File: modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/plugins/k8sjob/k8sjob.go [hunk 3]
```diff
@@ -499,6 +501,47 @@ func (k *K8sJob) generateKubeJob(specObj interface{}) (*batchv1.Job, error) {
...
+		if clusterInfo[apistructs.BuildkitHitRate] != "" {
+			hitRate, err = strconv.Atoi(clusterInfo[apistructs.BuildkitHitRate])
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse buildkit hit rate, err: %v", err)
+			}
+		}
+
+		rand.Seed(time.Now().UnixNano())
+
+		if rand.Intn(100) < hitRate {
```
Comment (sfwn) [inline modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/plugins/k8sjob/k8sjob.go:518]: extract a method and add ut for it.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Review by sixther-dc — COMMENTED** [review]
/lgtm

File: modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/plugins/k8sjob/k8sjob.go (hunk 2 above)
Comment (iutx) [reply modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/plugins/k8sjob/k8sjob.go:505]: Local variable operations do not require locking.

File: modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/plugins/k8sjob/k8sjob.go (hunk 3 above)
Comment (iutx) [reply modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/plugins/k8sjob/k8sjob.go]: ut result under 10000 times
```shell
k8sjob_test.go:37: rate: 0, result: 0
k8sjob_test.go:37: rate: 10, result: 9
k8sjob_test.go:37: rate: 50, result: 50
k8sjob_test.go:37: rate: 80, result: 79
k8sjob_test.go:37: rate: 100, result: 100
```

File: modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/plugins/k8sjob/k8sjob.go (hunk 1 above)
Comment (iutx) [reply modules/pipeline/pipengine/actionexecutor/plugins/scheduler/executor/plugins/k8sjob/k8sjob.go]: done

**Review by sixther-dc — APPROVED** [review]
/approve

**Review by sfwn — APPROVED** [review]

**Pull request merged**